### DISCUSSION TITLE
Исключения

### DISCUSSION THREAD

**Discussion by zhenyakoltsov** [bol-van/zapret#858] [Q&A]
Добрый день, настроил и все работает, только вот некоторые сайты работаю не корректно (vk.com и ok.ru не грузятся полностью), добавил данные адреса в файл zapret-hosts-user-exclude.txt и заново запустил файл install_easy.sh, в файле zapret-ip-exclude.txt появились новые ip адреса, но сайты все равно не грузятся. 
[config.txt](https://github.com/user-attachments/files/16491006/config.txt)
[zapret-hosts-user-exclude.txt](https://github.com/user-attachments/files/16491007/zapret-hosts-user-exclude.txt)

**Comment by k0zyrev** [comment]
Эти сайты используют по нескольку доменов для обеспечения своей работы, скорее всего, нужно добавить и их тоже. К каким доменам подключается сайт можно посмотреть через F12 (инструменты разработчика)>вкладка сеть. Если вы используете ublock origin, то он в расширенном режиме тоже показывает все подключенные домены.

Альтернативно, вы можете использовать парамет --hostlist="aaaaa.txt" и добавить в файл только те сайты, которые заблокированы или замедляются (как ютуб). Тогда все остальные точно не будут ломаться. Но в этом случае тоже нужно будет добавлять различные технические домены.

**Comment by bol-van** [comment]
добавьте в стратегию доя тцп (не куик)
--dpi-desync-fooling=md5sig

**Comment by zhenyakoltsov** [comment]
> Эти сайты используют по нескольку доменов для обеспечения своей работы, скорее всего, нужно добавить и их тоже. К каким доменам подключается сайт можно посмотреть через F12 (инструменты разработчика)>вкладка сеть. Если вы используете ublock origin, то он в расширенном режиме тоже показывает все подключенные домены.
> 
> Альтернативно, вы можете использовать парамет --hostlist="aaaaa.txt" и добавить в файл только те сайты, которые заблокированы или замедляются (как ютуб). Тогда все остальные точно не будут ломаться. Но в этом случае тоже нужно будет добавлять различные технические домены.

Данный параметр добавить в конец стратегии, должно получиться так: TPWS_OPT="--split-pos=1 --mss=88 --mss-pf=443 --hostlist="aaaaa.txt"? Где то еще нужно что то поменять в настройках?

**Comment by zhenyakoltsov** [comment]
> добавьте в стратегию доя тцп (не куик) --dpi-desync-fooling=md5sig

Данный параметр дописать в конец стратегии так: TPWS_OPT="--split-pos=1 --mss=88 --mss-pf=443  --dpi-desync-fooling=md5sig"?

**Comment by zhenyakoltsov** [comment]
Можно ли писать так: *vk.com что бы разблокировать сразу api.vk.com, im.vk.com, login.vk.com, queuev4.vk.com, st4-19.vk.com?

**Comment by bol-van** [comment]
параметры смешаны tpws и nfqws что есть чушь
читайте доку

на счет хостлистов субдомены учитываются автоматом
звезды не работают

**Comment by shuguroff** [comment]
Похожая проблема при применении параметров
--mss=88 --mss-pf=443

Не знаю куда копать))

github.com сперва полностью не открывался, хотя включен только белый список, в котором только ютюб. 

Далее я добавил github.com (и некоторые связанные домены) в список исключений, теперь сайт открывается, дает пройти пару страниц и перестает работать минут на 5. 

Если мы убираем mss, то все работает нормально. 

Пока писал выяснилось, что помимо скрипта get_user.sh есть еще get_exclude.sh, который я не выполнял.

Сейчас запущу и понаблюдаю, возможно максимально глупая ошибка с моей стороны)

В конфиге:

`
FWTYPE=nftables

MODE_FILTER=hostlist

TPWS_OPT="-hostspell=HoST --split-pos=1 --split-http-req=method --oob --split-pos=1 --mss=88 --mss-pf=443"
`

zapret-hosts-user.txt
`
googleapis.com
googletagmanager.com
googlevideo.com
i.ytimg.com
i9.ytimg.com
cdninstagram.com
gateway.instagram.com
graph.instagram.com
i-fallback.instagram.com
i.instagram.com
instagram.com
bcdn.net
instagram.fkun2-1.fna.fbcdn.net
instagram.frix7-1.fna.fbcdn.net
instagram.fvno2-1.fna.fbcdn.net
scontent-ams4-1.xx.fbcdn.net
cdninstagram.com
`

zapret-hosts-user-exclude.txt
`
10.0.0.0/8
172.16.0.0/12
192.168.0.0/16
169.254.0.0/16
fc00::/7
fe80::/10
github.com
accounts.github.com
api.github.com
collector.github.com
githubassets.com
jetbrains.com
account.jetbrains.com
`

**Comment by bol-van** [comment]
mss это очень жесткое издевательство
предназначено для пробива отдельных ресурсов
которые не поддерживают tls 1.3
для tls 1.3  это только вред

**Comment by shuguroff** [comment]
Никаких иных путей для моего провайдера не нашлось.

Суть в том, что при таком наборе параметров все работает, кроме гитхаба. А он находясь в исключениях работает через раз.

Как бы мне понять, что нужно сделать чтобы ну никак не применялся прокси к сайтам не из белого списка.

**Comment by bol-van** [comment]
снимайте процесс тпвс
перезапускайте с теми же параметрами в консоли с --debug
он все скажет что он творит

**Comment by shuguroff** [comment]
> снимайте процесс тпвс перезапускайте с теми же параметрами в консоли с --debug он все скажет что он творит

Я так понимаю программа сперва видит домен и, находит его в списке и никаких действий не предпринимает. А вот потом появляется второе соединение (уже по IP), которое почему-то не проверяется на наличие в списках  и пролетает мимо правил файрвола. 

Давайте еще подсказку, будем копать )

packet contains TLS ClientHello
hostname: github.com
Checking exclude hostlist
Hostlist check for github.com : positive
Not acting on this request
Socket fd=105 (partner_fd=106, remote=0) closed, connection removed. total_read=833 total_write=0 event_count=1
Legs : local:39 remote:40
Original destination for socket fd=105 : 140.82.121.4:443
Setting MSS 88
Legs : local:40 remote:41
Socket fd=105 (local) connected

**Comment by bol-van** [comment]
все понятно. mss работает на этапе установления тцп конекта
хост тогда еще неизвестен. фильтр по листу невозможен
но есть одно исключение
tpws в сокс режиме и клиент, использующий удаленный ресолв
 curl --socks5-hostname

**Comment by zhenyakoltsov** [comment]
Не могу понять как настроить что бы разблокировка шла только на определенные сайты через параметр --hostlist=имя_файла.txt, прилагаю файлы настроек, подскажите пожалуйста. Не понятно по параметру MODE_FILTER=none, нужно его менять?
[block.txt](https://github.com/user-attachments/files/16645327/block.txt)
[config.txt](https://github.com/user-attachments/files/16645328/config.txt)

**Comment by bol-van** [comment]
лучше прочитайте доку
хостлисты интегрированы в скрипты запуска

**Comment by zhenyakoltsov** [comment]
Пытался читать документацию, но ничего не понял подскажите в какой файл внести доменные имена что бы заработал данный метод.

**Comment by bol-van** [comment]
ipset/zapret-hosts-user.txt
MODE_FILTER=hostlist

**Comment by zhenyakoltsov** [comment]
Почему то у меня не работает, проверяю на сайте github.com который не работает при запущенном zapret.service, что бы я не менял сайт все равно не грузится при запущенном zapret.service, если отключить zapret.sirvice сайт грузится.

Содержимое zapret-hosts-user.txt:
rutor.info
ajax.googleapis.com
cdnbunny.or
sheisnotateacher.com
vak345.com
yadro.ru
counter.yadro.ru
rutracker.org
betsonsport.ru
www.betsonsport.ru
cloudflareinsights.com
static.cloudflareinsights.com
robinbob.in
rutracker.cc
static.rutracker.cc
rutrk.org
yadro.ru
counter.yadro.ru
nnmclub.to
google-analytics.com
ssl.google-analytics.com
www.google-analytics.com
googlesyndication.com
pagead2.googlesyndication.com
nnmstatic.win
yadro.ru
counter.yadro.ru
yandex.ru
mc.yandex.ru
ytimg.com
s.ytimg.com
lostfilm.tv
p1.lostfilm.tv
google-analytics.com
www.google-analytics.com
jsdelivr.net
cdn.jsdelivr.net
lostfilm.info
lostfilm.top
static.lostfilm.top
rambler.ru
counter.rambler.ru
tns-counter.ru
www.tns-counter.ru
yadro.ru
counter.yadro.ru
yandex.ru
mc.yandex.ru
yastatic.net
doramaclub.co
jsdelivr.net
cdn.jsdelivr.net
vak345.com
yadro.ru
counter.yadro.ru
doramy.club
googletagmanager.com
www.googletagmanager.com
jadekoi.ru
yadro.ru
counter.yadro.ru
youtube.com
accounts.youtube.com
suggestqueries-clients6.youtube.com
www.youtube.com
doubleclick.net
googleads.g.doubleclick.net
static.doubleclick.net
fonts.googleapis.com
ggpht.com
yt3.ggpht.com
google.com
www.google.ru
googlevideo.com
rr1---sn-xguxaxjvh-8v1e.googlevideo.com
rr1---sn-xguxaxjvh-8v1s.googlevideo.com
rr2---sn-xguxaxjvh-8v1e.googlevideo.com
rr3---sn-xguxaxjvh-8v1e.googlevideo.com
rr4---sn-q4flrner.googlevideo.com
rr5---sn-xguxaxjvh-8v1e.googlevideo.com
gstatic.com
fonts.gstatic.com
www.gstatic.com
jnn-pa.googleapis.com
ytimg.com
i.ytimg.com
x.com
ads-twitter.com
cdn-apple.com
google.com
googleusercontent.com
gstatic.com
twimg.com
facebook.com
www.facebook.com
fbcdn.net

Содержимое zapret-ip-user.txt:
104.16.79.73
104.16.80.73
104.21.17.86
104.21.31.9
104.21.32.39
104.21.50.150
104.21.73.130
104.21.96.59
104.244.42.1
104.244.42.129
104.244.42.193
104.244.42.65
1.2.3.4
142.250.217.100
142.250.217.106
142.250.217.110
142.250.217.118
142.250.217.74
142.250.217.86
142.250.69.193
142.250.69.194
142.250.69.198
142.250.69.202
142.250.69.206
142.250.69.214
142.251.211.227
142.251.211.232
142.251.211.234
142.251.211.246
142.251.215.227
142.251.215.234
142.251.215.238
142.251.215.246
142.251.33.106
142.251.33.118
142.251.33.65
142.251.33.74
142.251.33.78
142.251.33.86
151.101.1.229
151.101.129.229
151.101.193.229
151.101.65.229
157.240.201.35
157.240.205.35
172.217.131.9
172.217.14.196
172.217.14.202
172.217.14.206
172.67.162.234
172.67.163.237
172.67.173.153
172.67.174.156
172.67.175.99
172.67.182.196
173.194.221.94
173.194.73.100
173.194.73.101
173.194.73.102
173.194.73.113
173.194.73.138
173.194.73.139
178.154.131.215
178.154.131.217
185.45.144.60
188.114.96.1
188.114.96.2
188.114.97.1
188.114.97.2
188.234.138.172
188.234.140.76
188.234.140.77
188.234.140.78
188.234.140.80
193.200.64.24
193.46.255.29
194.226.130.226
194.226.130.227
194.226.130.228
194.226.130.229
195.24.66.175
198.211.127.57
199.232.169.229
216.239.32.178
216.239.34.178
216.239.36.178
216.239.38.178
5.255.255.77
64.233.162.120
64.233.162.198
64.233.162.94
64.233.164.154
64.233.164.155
64.233.164.156
64.233.164.157
64.233.165.97
74.125.205.94
77.88.21.119
77.88.44.55
77.88.55.88
81.19.82.0
81.19.82.1
81.19.82.2
81.19.89.16
81.19.89.17
81.19.89.18
87.242.104.43
87.250.250.119
87.250.251.119
88.212.196.78
88.212.201.198
88.212.201.204
88.212.202.52
93.158.134.119

Содержимое файла config приложено ниже.
[config.txt](https://github.com/user-attachments/files/16647401/config.txt)

Файлы zapret-hosts-user-exclude.txt и zapret-ip-exclude.txt пустые.

вывод journalctl -u zapret service:
авг 18 13:27:38 benq systemd[1]: Starting zapret.service...
авг 18 13:27:38 benq zapret[1575]: Starting daemon 1: /opt/zapret/tpws/tpws --user=tpws --bind-addr=127.0.0.127 --port=988 --split-pos=1 --mss=88 --mss-pf=443 --hostlist=/opt/zapret/ipset/zapret-hosts-user.txt --hostlist-exclude=/opt/z>
авг 18 13:27:38 benq zapret[1575]: Applying iptables
авг 18 13:27:38 benq zapret[1575]: Creating ip list table (firewall type iptables)
авг 18 13:27:38 benq zapret[1602]: setting high oom kill priority
авг 18 13:27:38 benq zapret[1602]: reloading ipset backend (no-update)
авг 18 13:27:38 benq zapret[1602]: Adding to ipset zapret : /opt/zapret/ipset/zapret-ip-user.txt
авг 18 13:27:38 benq zapret[1602]: Adding to ipset ipban : /opt/zapret/ipset/zapret-ip-user-ipban.txt
авг 18 13:27:38 benq zapret[1602]: Adding to ipset nozapret : /opt/zapret/ipset/zapret-ip-exclude.txt
авг 18 13:27:38 benq zapret[1575]: Adding iptables rule for tpws (port 988) :  -p tcp -m multiport --dports 80,443
авг 18 13:27:38 benq systemd[1]: Started zapret.service.

**Comment by bol-van** [comment]
mss и syndata игнорируют хостлист

**Comment by bol-van** [comment]
если у вас виртуалка с нат, гипервизор сломает большинство методов обхода

**Comment by zhenyakoltsov** [comment]
У меня не виртуалка, получается единственный способ добавить сайт github в файл zapret-hosts-user-exclude.txt, я правильно понимаю?

**Comment by bol-van** [comment]
с мсс будет работать только ип фильтр
никакие хостлисты не сработают

**Comment by shuguroff** [comment]
Вы зря боретесь с ускорение ютуба через mss, я тоже шел этим путем и ломал все остальное

По итогу в теме https://github.com/bol-van/zapret/discussions/200 был найден другой набор параметров, там же объясняется, что blockcheck для поиска вариантов обхода в этом случае бесполезен

**Comment by 0Eagle** [comment]
> Добрый день, настроил и все работает, только вот некоторые сайты работаю не корректно (vk.com и ok.ru не грузятся полностью), добавил данные адреса в файл zapret-hosts-user-exclude.txt и заново запустил файл install_easy.sh, в файле zapret-ip-exclude.txt появились новые ip адреса, но сайты все равно не грузятся. [config.txt](https://github.com/user-attachments/files/16491006/config.txt) [zapret-hosts-user-exclude.txt](https://github.com/user-attachments/files/16491007/zapret-hosts-user-exclude.txt)

и как этот файл с исключениями прописать в батник под windows? в доках ни слова, догадайтесь сами...

**Comment by bol-van** [comment]
В доках описана работа с хостлистами.
Только придется их делать самому и указывать в параметрах тоже самому , причем в правильном месте

**Comment by AntonSamokat** [comment]
@bol-van Как применить изменённый файл `zapret-hosts-user-exclude.txt`?  Можно запускать скрипт `create_ipset.sh` при работающем сервисе? Или лучше выполнять `install_easy.sh`?

---

У меня была проблема с обновлениями для Ubuntu при работающем сервисе. Серверы apt не отвечали:
```
sudo apt update
Hit:1 http://archive.ubuntu.com/ubuntu focal InRelease                                    
Err:2 http://archive.ubuntu.com/ubuntu focal-updates InRelease
  400  Bad Request [IP: 91.189.91.81 80]
0% [Waiting for headers]^C
```

Добавил их в zapret-hosts-user-exclude.txt:
```
cat /opt/zapret/ipset/zapret-hosts-user-exclude.txt
10.0.0.0/8
172.16.0.0/12
192.168.0.0/16
169.254.0.0/16
fc00::/7
fe80::/10
archive.ubuntu.com
archive.canonical.com
```

Выполнил `install_easy.sh` с выбором filtering = ipset, download ip/host list = yes, далее всё по умолчанию.
```
select filtering :
1 : none
2 : ipset
3 : hostlist
4 : autohostlist
your choice (default : none) : 2
selected : ipset

do you want to auto download ip/host list (default : N) (Y/N) ? y
1 : get_user.sh
2 : get_refilter_ipsum.sh
3 : get_antifilter_ip.sh
4 : get_antifilter_ipsmart.sh
5 : get_antifilter_ipsum.sh
6 : get_antifilter_ipresolve.sh
7 : get_antifilter_allyouneed.sh
8 : get_reestr_resolve.sh
9 : get_reestr_preresolved.sh
10 : get_reestr_preresolved_smart.sh
your choice (default : get_antifilter_allyouneed.sh) : 
selected : get_antifilter_allyouneed.sh
* installing zapret service
Created symlink /etc/systemd/system/multi-user.target.wants/zapret.service → /lib/systemd/system/zapret.service.
* downloading blocked ip/host list
setting high oom kill priority
clearing all known DNS caches
DNS is working
digging 8 ipv4 domains : /opt/zapret/ipset/zapret-hosts-user-exclude.txt
mdig stats : 00:00:05 : domains=8 success=6 error=2
digging 1 ipv4 domains : /opt/zapret/ipset/zapret-hosts-user.txt
mdig stats : 00:00:00 : domains=1 success=0 error=1
digging 0 ipv4 domains : /opt/zapret/ipset/zapret-hosts-user-ipban.txt
mdig stats : 00:00:00 : domains=0 success=0 error=0
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  171k  100  171k    0     0  64729      0  0:00:02  0:00:02 --:--:-- 64730
setting high oom kill priority
reloading ipset backend (forced-update)
Adding to ipset zapret : /opt/zapret/ipset/zapret-ip.txt
Adding to ipset zapret : /opt/zapret/ipset/zapret-ip-user.txt
Adding to ipset ipban : /opt/zapret/ipset/zapret-ip-user-ipban.txt
Adding to ipset nozapret : /opt/zapret/ipset/zapret-ip-exclude.txt
* installing zapret-list-update timer
Removed /etc/systemd/system/timers.target.wants/zapret-list-update.timer.
Created symlink /etc/systemd/system/timers.target.wants/zapret-list-update.timer → /lib/systemd/system/zapret-list-update.timer.
* starting zapret service

press enter to continue
```

Коннект к apt серверам появился.
